core/src/main/java/org/apache/iceberg/inmemory/InMemoryCatalog.java

```diff
@@ -287,6 +294,69 @@ public List<Namespace> listNamespaces(Namespace namespace) throws NoSuchNamespac
   public void close() throws IOException {
     namespaces.clear();
     tables.clear();
+    views.clear();
+  }
+
+  @Override
+  public List<TableIdentifier> listViews(Namespace namespace) {
+    if (!namespaceExists(namespace) && !namespace.isEmpty()) {
+      throw new NoSuchNamespaceException(
+          "Cannot list views for namespace. Namespace does not exist: %s", namespace);
+    }
+
+    return views.keySet().stream()
+        .filter(t -> namespace.isEmpty() || t.namespace().equals(namespace))
+        .sorted(Comparator.comparing(TableIdentifier::toString))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public View loadView(TableIdentifier identifier) {
+    if (isValidIdentifier(identifier) && views.containsKey(identifier)) {
+      return new BaseView(newViewOps(identifier), fullTableName(name(), identifier));
```

Comment:
For views, do we want to continue using the String name or should we return an identifier?

Reply:
I've added the catalog name and the identifier to `BaseView` and `BaseView#identifier()` returns the identifier while `BaseView#name()` returns the fully-qualified name that includes the catalog name.

Not sure if it's worth, but we could introduce a `FullIdentifier` class that would hold the original catalog name and the underlying `TableIdentifier` (which could also be helpful for #7419). With that we could gradually move away from the plain string to `FullIdentifier`.